firedrake/assemble.py

```diff
@@ -231,8 +231,8 @@ def vec(testmap, i):
     # assemble it.
     def thunk(bcs):
         zero_tensor()
-        for (i, j), integral_type, subdomain_id, coords, coefficients, needs_orientations, kernel in kernels:
-            m = coords.function_space().mesh()
+        for (i, j), integral_type, m, subdomain_data, subdomain_id, coefficients, needs_orientations, kernel in kernels:
+            coords = m.coordinates
             if needs_orientations:
```

Comment:
Can you raise a NotImplementedError here if subdomain_data is provided and the integral_type is not `cell`?